src/test/mir-opt/funky_arms/rustc.float_to_exponential_common.ConstProp.diff

```diff
@@ -0,0 +1,158 @@
+      let mut _13: core::num::flt2dec::Sign; // in scope 0 at $DIR/funky_arms.rs:23:53: 23:57
+      let mut _14: usize;                  // in scope 0 at $DIR/funky_arms.rs:23:59: 23:72
+      let mut _15: usize;                  // in scope 0 at $DIR/funky_arms.rs:23:59: 23:68
+      let mut _16: bool;                   // in scope 0 at $DIR/funky_arms.rs:23:74: 23:79
+      let mut _17: &mut std::fmt::Formatter; // in scope 0 at $DIR/funky_arms.rs:25:46: 25:49
+      let mut _18: &T;                     // in scope 0 at $DIR/funky_arms.rs:25:51: 25:54
+      let mut _19: core::num::flt2dec::Sign; // in scope 0 at $DIR/funky_arms.rs:25:56: 25:60
+      let mut _20: bool;                   // in scope 0 at $DIR/funky_arms.rs:25:62: 25:67
+      scope 1 {
+          debug force_sign => _4;          // in scope 1 at $DIR/funky_arms.rs:15:9: 15:19
+          let _6: core::num::flt2dec::Sign; // in scope 1 at $DIR/funky_arms.rs:16:9: 16:13
+          scope 2 {
+              debug sign => _6;            // in scope 2 at $DIR/funky_arms.rs:16:9: 16:13
+              let _10: usize;              // in scope 2 at $DIR/funky_arms.rs:21:17: 21:26
+              scope 3 {
+                  debug precision => _10;  // in scope 3 at $DIR/funky_arms.rs:21:17: 21:26
+              }
+          }
+      }
+  
+      bb0: {
+          StorageLive(_4);                 // scope 0 at $DIR/funky_arms.rs:15:9: 15:19
+          StorageLive(_5);                 // scope 0 at $DIR/funky_arms.rs:15:22: 15:25
+          _5 = &(*_1);                     // scope 0 at $DIR/funky_arms.rs:15:22: 15:25
+          _4 = const std::fmt::Formatter::sign_plus(move _5) -> bb1; // scope 0 at $DIR/funky_arms.rs:15:22: 15:37
+                                           // ty::Const
+                                           // + ty: for<'r> fn(&'r std::fmt::Formatter) -> bool {std::fmt::Formatter::sign_plus}
+                                           // + val: Value(Scalar(<ZST>))
+                                           // mir::Constant
+                                           // + span: $DIR/funky_arms.rs:15:26: 15:35
+                                           // + literal: Const { ty: for<'r> fn(&'r std::fmt::Formatter) -> bool {std::fmt::Formatter::sign_plus}, val: Value(Scalar(<ZST>)) }
+      }
+  
+      bb1: {
+          StorageDead(_5);                 // scope 0 at $DIR/funky_arms.rs:15:36: 15:37
+          StorageLive(_6);                 // scope 1 at $DIR/funky_arms.rs:16:9: 16:13
+          switchInt(_4) -> [false: bb3, otherwise: bb2]; // scope 1 at $DIR/funky_arms.rs:17:9: 17:14
+      }
+  
+      bb2: {
+          discriminant(_6) = 2;            // scope 1 at $DIR/funky_arms.rs:18:17: 18:41
+          goto -> bb4;                     // scope 1 at $DIR/funky_arms.rs:16:16: 19:6
+      }
+  
+      bb3: {
+          discriminant(_6) = 0;            // scope 1 at $DIR/funky_arms.rs:17:18: 17:38
```

Comment:
@wesleywiser the bug used to be that this (the last assignment to `_6` in this MIR) assigned a constant to `_6` and didn't erase it at the end of the block.